src/mbgl/layout/symbol_layout.cpp

```diff
@@ -637,26 +642,25 @@ void SymbolLayout::createBucket(const ImagePositions&, std::unique_ptr<FeatureIn
         // Process icon first, so that text symbols would have reference to iconIndex which
         // is used when dynamic vertices for icon-text-fit image have to be updated.
         if (hasIcon) {
-            if (symbolInstance.hasIcon) {
```

Comment:
symbol instance might have no icon. Could we have render test for that? Symbol layer with data driven icon-image that will be empty for one of the instances.